marconi-core/test/Marconi/CoreSpec.hs

```diff
@@ -1636,6 +1706,74 @@ withRollbackFailureTest =
     Test.withMaxSuccess 10_000 $
       checkOutOfRollback (Lens.view defaultChain <$> Test.arbitrary)
 
+-- * withStream
+
+withStreamTestGroup :: Tasty.TestTree
+withStreamTestGroup =
+  Tasty.testGroup
+    "StreamTests"
+    [ Tasty.testGroup
+        "Stream"
+        [ Tasty.testGroup
+            "withStreamTBQueue"
+            [ Tasty.testProperty "TBQueue streaming works" withStreamTBQueue
+            ]
+        , Tasty.testGroup
+            "withStreamSocket"
+            [ Tasty.testProperty "Socket streaming works" withStreamSocket
+            ]
+        ]
+    ]
+
+withStreamTBQueue :: Property
```

Comment:
Property tests should start with `prop`. Also add a comment for each explaining _what_ your testing exactly.